source/pymongo-async-migration.txt

```diff
@@ -0,0 +1,358 @@
+.. _pymongo-async-migration:
+
+===================================
+Migrate to the {+driver-async+} Driver
+===================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: motor, async, refactor, migration
+
+Overview
+--------
+
+In September 2024, MongoDB released the {+driver-async+} driver to unify {+driver-short+}
+and `Motor <https://www.mongodb.com/docs/drivers/motor/>`__, the asynchronous
+MongoDB driver for Python. In this guide, you can identify the changes you must
+make to migrate an application from {+driver-short+} or Motor to the
+{+driver-async+} driver.
+
+Migrate From {+driver-short+}
+--------------------
+
+The {+driver-async+} driver behaves similarly to {+driver-short+}, but
+all methods that perform network operations are coroutines and must be awaited.
+To migrate from {+driver-short+} to {+driver-async+}, you must update your code
+in the following ways:
+
+- Replace all uses of ``MongoClient`` with ``AsyncMongoClient``.
+- Add the ``await`` keyword to all asynchronous method calls .
+- If an asynchronous method is called within a function, mark the function as ``async``.
+
+The following sections describe how to implement the asynchronous API.
+
+Asynchronous Methods
+~~~~~~~~~~~~~~~~~~~~
+
+The following tables list the asynchronous methods that are available in the
+{+driver-async+} driver. To call these methods, you must ``await`` them and call them
+inside an ``async`` function.
+
+Client Methods
+``````````````
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 80
+
+   * - Method
+     - Example
+
+   * - ``AsyncMongoClient()``
+     - .. code-block:: python
+
+          from pymongo import AsyncMongoClient
+
+          client = AsyncMongoClient(...)
+
+   * - ``watch()``
+     - .. code-block:: python
+
+          async with await client.watch(...) as stream:
+          ...
+
+   * - ``server_info()``
+     - .. code-block:: python
+
+          info = await client.server_info(...)
```

Comment:
Q: did you only assign values for methods that have a return type? I wonder if it would be more simple to just omit the assignment variable from all of the examples

Reply:
I took this from a list of methods that were probably all taken from random areas so I don't think there was an exact methodology. I removed them to make it simpler